### PR TITLE
Updating home.py and adding test class for it

### PR DESCRIPTION
**Pull request by brkykrclk**
home,py updated and working now, some updates needs for future
test class only for connection so far

### CODE REVIEW

File: src/flaskr/home.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        # Initialize DataGeneration with form data
+        try:
+            data_gen = DataGeneration(channel_id, time_increment, field_id, start_date)
+            date_series, value_series = data_gen.get_time_series()
+        except Exception as e:
+            flash(f"Error while generating data: {e}")  # Use flash for error messages
+            return redirect(url_for('home'))
+
+        # Initialize Sensor with fetched data and pass data to the template for rendering
+        sensor = Sensor(name="Generated Sensor", description="Data from ThingSpeak", data={'dates': date_series, 'values': value_series})
```
Comment (Jcp149) [inline src/flaskr/home.py:29]: i dont think this is how a sensor is made

File: src/flaskr/home.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        return render_template('main/home.html', labels=date_series, values=value_series)
```
Comment (Jcp149) [inline src/flaskr/home.py:30]: your home doesnt have the chart stuff implemented for this

**Review by Birdbh — CHANGES_REQUESTED** [review]
@Jcp149 comments are correct, you are not calling the creating the sensor object correctly. This leads to an error at runtime:

image

Here is an image of what happens when the submit button is clicked because of this improper creation. Here is the error:
File "C:\Users\birdl\Desktop\Year 4\Term 2\COSC 310\Workspace\Engicoders\src\flaskr\home.py", line 29, in home
    sensor = Sensor(name="Generated Sensor", description="Data from ThingSpeak", data={'dates': date_series, 'values': value_series})
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Sensor.__init__() got an unexpected keyword argument 'data'

Please review my work, if you need reference

**Review by Birdbh — APPROVED** [review]
I did some user testing:
image
The data displays with public data, and will updated when a new private channel is added -> This is good

The one To Do: in the future will be an error or some form of response for when a non-public field is selected

**Pull request merged**